Cause scalar string with colon to be quoted #380

darrelmiller opened the pull request:
A trailing colon in a string causes the YAML parser to think the scaler is a map key.  This change ensures scalars with colons are quoted.  Arguably, we could allow colons in the middle of strings as long as they are not followed by a space but, this change should be sufficient.  Fixes #380

PerthCharern commented on src/Microsoft.OpenApi/Writers/SpecialCharacterStringExtensions.cs:
```diff
@@ -42,7 +42,8 @@ public static class SpecialCharacterStringExtensions
         private static readonly string[] _yamlPlainStringForbiddenCombinations =
         {
-            ": ",
+            //": ",
```
nit, remove this commented-out line

PerthCharern commented on src/Microsoft.OpenApi/Writers/SpecialCharacterStringExtensions.cs:
```diff
@@ -42,7 +42,8 @@ public static class SpecialCharacterStringExtensions
         {
-            ": ",
+            //": ",
+            ":",  // Even though colons can be allowed, this change prevents unquoted trailing colon which causes a problem
```
Even though colons not followed by a space can be allowed, this change prevents unquoted trailing colon which causes a problem.

PerthCharern commented:
@darrelmiller

- Though this works, this will add unnecessary single quotes for a frequent usage of URL (http://example.com) since it contains a colon in the scheme. Can we create a different list of "forbidden characters at end of string" and include colon in it instead? That would add the single quotes for this specific case while still keeping the URL case as normal.

- Unit tests

darrelmiller commented:
@PerthCharern That makes sense.  I try and find time to work on that this week, although that might be a bit optimistic :-)

darrelmiller commented:
@PerthCharern I updated this PR to check for colon as a terminal character.  I have been unable to find other characters that cause a problem.  I added a test for the scenario
